compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -628,14 +735,29 @@ pub(crate) unsafe fn optimize(
         unsafe { llvm::LLVMWriteBitcodeToFile(llmod, out.as_ptr()) };
     }
 
+    // FIXME(ZuseZ4): support SanitizeHWAddress and prevent illegal/unsupported opts
+
     if let Some(opt_level) = config.opt_level {
         let opt_stage = match cgcx.lto {
             Lto::Fat => llvm::OptStage::PreLinkFatLTO,
             Lto::Thin | Lto::ThinLocal => llvm::OptStage::PreLinkThinLTO,
             _ if cgcx.opts.cg.linker_plugin_lto.enabled() => llvm::OptStage::PreLinkThinLTO,
             _ => llvm::OptStage::PreLinkNoLTO,
         };
-        return unsafe { llvm_optimize(cgcx, dcx, module, config, opt_level, opt_stage) };
+
+        // If we know that we will later run AD, then we disable vectorization and loop unrolling
+        let skip_size_increasing_opts = cfg!(llvm_enzyme);
```

Comment:
Shouldn't this just be `true` like at the other site, considering llvm_optimize internally checks the cfg again?

Reply:
It could, but then readers not knowing about AD would be left wondering why we skip them (and need to check the code to find out that we actually don't). It also looks a bit strange because I removed some of the config options in the first upstreaming pass (we'll have a few more debug options later). I don't care too strongly either way (and as mentioned in the comment, would prefer to make it more granular in the future anyway), so I can change it if you still want?